lib/react_client/react_interop.dart

```diff
@@ -89,15 +96,20 @@ class Ref<T> {
   T get current {
     final jsCurrent = jsRef.current;
 
-    if (jsCurrent is! Element) {
-      final dartCurrent = (jsCurrent as ReactComponent)?.dartComponent;
+    if (jsCurrent is! Element && jsCurrent is ReactComponent) {
+      final dartCurrent = jsCurrent?.dartComponent;
 
       if (dartCurrent != null) {
         return dartCurrent as T;
       }
     }
     return jsCurrent;
   }
+
+  /// Sets the value of [current].
+  ///
+  /// See: <https://reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables>.
+  set current(dynamic value) => jsRef.current = value;
```

Comment:
This should be `T`, so they can't statically set an incompatible value
```suggestion
  set current(T value) => jsRef.current = value;
```